src/models/embed/index.ts

```diff
@@ -88,7 +88,12 @@ export class EmbeddingsTask {
 
     while (!isDone()) {
       await this.sleep(sleepInterval);
-      await this.updateStatus();
+      try {
+        await this.updateStatus();
```

Comment:
as you update the status, I think it'll be nice to mutate `this.status` (inside `this.updateStatus`)

Reply:
nice catch :) updated method to easily understand